superset/assets/spec/javascripts/explore/AdhocFilter_spec.js

```diff
@@ -0,0 +1,136 @@
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+
+import AdhocFilter, { EXPRESSION_TYPES, CLAUSES } from '../../../src/explore/AdhocFilter';
+
+describe('AdhocFilter', () => {
+  it('sets filterOptionName in constructor', () => {
+    const adhocFilter = new AdhocFilter({
+      expressionType: EXPRESSION_TYPES.SIMPLE,
+      subject: 'value',
+      operator: '>',
+      comparator: '10',
+      clause: CLAUSES.WHERE,
+    });
+    expect(adhocFilter.filterOptionName.length).to.be.above(10);
```

Comment:
is this because `value > 10` has a length of 10? you could maybe make it more readable/obvious by doing `'value > 10'.length`

Reply:
ah the optionNames are just randomized ids in this case. we use them to differentiate options when updating (sort of like a react key). so I can't predict what its value is.